### ISSUE TITLE
Scrolling Performance is slow in Android and iOS.

### ISSUE THREAD

**Issue by AneesFathimaS** [dotnet/maui#25080]
### Description

1.	When using the collection view in three columns of the Grid the scrolling performance is slow in ios. While I use it in one column the scrolling performance is somewhat better.
2.	When I use Bindable Stacklayout inside the ScrollView the output itself not shown. It takes more time to load.

### Steps to Reproduce

1.	Click on the first button "CollectionView in three Columns".
2.	Scroll the three columns and check the scrolling performance is slow.
3.	Click on the second button "CollectionView - one column".
4.	Scroll and check the scrolling performance is somewhat better.
5.	Click on the third button "Stack BindableLayout in three columns" it taking more time to load and the output is not shown.
6.	Click on the fourth button "Stack BindableLayout - one column" it takes more time to load and the output is not shown.

### Link to public reproduction project repository

[MauiSample.zip](https://github.com/user-attachments/files/17255230/MauiSample.zip)

### Version with bug

8.0.91 SR9.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

**Comment by jfversluis** [comment]
Maybe you can share a .speedscope file of a Release build of your app in action?

https://aka.ms/profile-maui

That would help us find problem areas to improve.

**Comment by jonmdev** [comment]
> Maybe you can share a .speedscope file of a Release build of your app in action?
> 
> https://aka.ms/profile-maui
> 
> That would help us find problem areas to improve.

I do not think this is probable. There are already numerous already identified issues with the Layout system which will explain poor scrolling performance. These are identified across many bug report threads. I am not sure how a speedscope would help solve any of the already known issues.

Speed will not increase unless all the added unnecessary draws, measures, and arranges are pruned out of the Layout system and the same number are being performed per load or scroll as in Xamarin. I can imagine no way around this reality.

**Comment by Zhanglirong-Winnie** [comment]
This issue has been verified using  Visual Studio 17.12.0 Preview 3(8.0.92& & 8.0.91 & 8.0.90). Can repro this issue on android and iOS platforms.

**Comment by alpayclsr1** [comment]
We have started experiencing this issue as well. We have two Mac devices—one updated to Sequoia. When we run the app on the device with Maui version 8.0.92, the CollectionView scrolls so poorly that the app becomes unusable on iPad with iOS 18. However, when we take the IPA from a device that hasn't been updated to Sequoia and is running Maui version 8.0.40, everything works perfectly on the iOS 18 device. The CollectionView scrolling is very smooth. @jfversluis

**Comment by jsuarezruiz** [comment]
cc @jonathanpeppers

**Comment by Shalini-Ashokan** [comment]
The issue was checked in Visual Studio Code 1.103.2 with .NET MAUI version 9.0.100 in Release mode. Scrolling performance is better on iOS for both CV1 and CV2, and a video is attached for reference.
 
**Note**: On Android, the scrolling performance is still poor, and the issue is replicated.

https://github.com/user-attachments/assets/c8e484e8-e859-498f-935c-4ffcc7483790

https://github.com/user-attachments/assets/c06824cf-06a7-40ae-89f7-8e449c69069c
